h2o-core/src/main/java/water/webserver/H2OHttpViewImpl.java

```diff
@@ -68,16 +68,18 @@ public boolean authenticationHandler(HttpServletRequest request, HttpServletResp
       //TODO for LoginType.HASH, this equals not adding the handler at all; consinder this optimization
       return false;
     }
-
-    if (request.getUserPrincipal() == null) {
-      return false;
-    }
-    final String loginName = request.getUserPrincipal().getName();
-    if (loginName.equals(config.user_name)) {
-      return false;
+    
+    if (request.getUserPrincipal() != null) {
+      String loginName = request.getUserPrincipal().getName();
+      if (loginName.equals(config.user_name)) {
+        return false;
+      } else {
+        Log.warn("Login name (" + loginName + ") does not match cluster owner name (" + config.user_name + ")");
+        ServletUtils.sendResponseError(response, HttpServletResponse.SC_UNAUTHORIZED, "Login name does not match cluster owner name");
+      }
+    } else {
+      ServletUtils.sendResponseError(response, HttpServletResponse.SC_UNAUTHORIZED, "Missing authentication details");
```

Comment:
@honzasterba why is this needed? This handler should only be concerned with authorization = admit only users who are allowed to use the cluster. In the case of hash-authentication - anybody who is the config file, in case of other methods - only the user who started the cluster.